### PR TITLE
Set `filesize` and `duration` overflowed values to `None`

### LINKED ISSUE #877: Set `duration` and `filesize` values to `None` if they exceed integer maximum
## Context

See [Handling very large (>2GB) files](https://make.wordpress.org/openverse/2022/10/25/handling-very-large-2gb-files/), [Preparing the next migration of the Catalog](https://make.wordpress.org/openverse/2022/11/18/preparing-the-next-migration-of-the-catalog/), #730, #873

## Description

We would like to add some logic to the `MediaStore` base and/or derivatives to set the `filesize` and `duration` values to `None` if the value exceeds the maximum value for `integer`s in Postgres, 2147483647.

**Note**: This logic should be removed (and should be noted as such within the code) once a migration of the columns to type `bigint` is complete.

We could probably make a utility function on `MediaStore` which checks if a value is greater than the int maximum and returns `None` if it is, then apply that to `filesize` on the base class and `duration` on the `AudioStore` subclass.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #877 by @AetherUnbound

## Description

This PR adds a new validation function to the `MediaStore` class which is used to ensure that both `filesize` and `duration` do not exceed the Postgres integer maximum value.

I'm hoping I put the validation in a reasonable enough place, but if folks who have worked with the stores before (CC @obulat) have other suggestions I'm open to it!

## Testing Instructions

1. Run the tests locally
1. Optionally, ensure that the reproduction criteria defined in #730 succeeds under this PR

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/common/storage/media.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 PROVIDER_API = "provider_api"
 
 FILETYPE_EQUIVALENTS = {"jpeg": "jpg", "tif": "tiff"}
+INTEGER_MAXIMUM = 2147483647
```
Comment (krysal) [inline openverse_catalog/dags/common/storage/media.py:39]: Perhaps a prefix will make it more specific and clear! but I leave it up to you.

```suggestion
PG_INTEGER_MAXIMUM = 2147483647
```

**Review by krysal — APPROVED** [review]
LGTM

File: openverse_catalog/dags/common/storage/media.py (hunk 1 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/common/storage/media.py:39]: Oh that's a great call 😄

File: openverse_catalog/dags/common/storage/media.py [hunk 2]
```diff
@@ -288,6 +290,20 @@ def _validate_filetype(self, filetype: str | None, url: str) -> str | None:
             return filetype
         return FILETYPE_EQUIVALENTS.get(filetype, filetype)
 
+    @staticmethod
+    def _validate_integer(value: int | None) -> int | None:
+        """
+        Checks to ensure that a provided integer value is less than the Postgres
+        maximum integer value. If the value exceeds this maximum, None is returned.
+        TODO: Remove this logic once the column has been embiggened
```
Comment (stacimc) [inline openverse_catalog/dags/common/storage/media.py:298]: 😄

**Review by stacimc — APPROVED** [review]
This LGTM. As far as where to put the logic, this makes sense to me, especially since it will be removed. I suppose you could put it in the `IntegerColumn` itself? Personally I'm fine with the current implementation.

**Comment by AetherUnbound** [comment]
> I suppose you could put it in the `IntegerColumn` itself? Personally I'm fine with the current implementation.

Ah, that probably would have been a better implementation 😅 That said, since this is transient, I'll leave as-is.

**Pull request merged**